design-principles.md

```diff
@@ -31,7 +82,14 @@ When you do implement a feature, implement it the right way. Avoid
 workarounds. Workarounds merely kick the problem further down the
 road, but at a higher cost: someone will have to relearn the problem,
 figure out the workaround and how to dismantle it (and all the places
-that now use it), _and_ implement the feature.
+that now use it), _and_ implement the feature. It's much better to
+take longer to fix a problem properly, than to be the one who fixes
+everything quickly but in a way that will require cleaning up later.
+
+You may hear team members say "embrace the yak shave!". This is an
```

Comment:
i'll link to a definition, but really this whole paragraph is just trying to explain the term, so...